include/VectorGraph.h

```diff
@@ -0,0 +1,739 @@
+	// that are not used by points (there should be 0 cases like this)
+	void delUnusedAutomation();
+	QString getSavedDataArray();
+	void loadDataArray(QString data, unsigned int arraySize);
+private:
+	class VectorGraphPoint
+	{
+	public:
+		inline VectorGraphPoint()
+		{
+			m_x = 0.0f;
+			m_y = 0.0f;
+			m_c = 0.0f;
+			m_valA = 0.0f;
+			m_valB = 0.0f;
+			m_type = 0;
+			m_automatedEffectedAttribLocations = 0;
+
+			m_effectOnlyPoints = false;
+			m_effectAdd = false;
+			m_effectSubtract = false;
+			m_effectMultiply = false;
+			m_effectDivide = false;
+			m_effectPower = false;
+			m_effectLog = false;
+			m_effectSine = false;
+			m_effectClampLower = false;
+			m_effectClampUpper = false;
+
+			m_bufferedAutomationValue = 0.0f;
+			m_automationModel = -1;
+		}
+		inline VectorGraphPoint(float x, float y)
+		{
+			m_x = x;
+			m_y = y;
+			m_c = 0.0f;
+			m_valA = 0.0f;
+			m_valB = 0.0f;
+			m_type = 0;
+			m_automatedEffectedAttribLocations = 0;
+
+			m_effectOnlyPoints = false;
+			m_effectAdd = false;
+			m_effectSubtract = false;
+			m_effectMultiply = false;
+			m_effectDivide = false;
+			m_effectPower = false;
+			m_effectLog = false;
+			m_effectSine = false;
+			m_effectClampLower = false;
+			m_effectClampUpper = false;
+
+			m_bufferedAutomationValue = 0.0f;
+			m_automationModel = -1;
+		}
+		inline ~VectorGraphPoint()
+		{
+		}
+		// 0 - 1
+		float m_x;
+		// 0 (or -1) - 1, getAutomatedAttrib() -> 0
+		float m_y;
+		// curve, -1 - 1, getAutomatedAttrib() -> 1
+		float m_c;
+		// valueA, -1 - 1, getAutomatedAttrib() -> 2
+		float m_valA;
+		// valueB, -1 - 1, getAutomatedAttrib() -> 3
+		float m_valB;
+		// line type:
+		// 0 - none
+		// 1 - sine
+		// 2 - sineB
+		// 3 - peak
+		// 4 - steps
+		// 5 - random
+		unsigned int m_type;
+		// the automated attrib location and
+		// the effected attrib location is
+		// stored here
+		// use getAutomatedAttrib or getEffectedAttrib to get it
+		unsigned int m_automatedEffectedAttribLocations;
+
+		// if the point's line should effect only points
+		// getEffectOnlyPoints() will return true when
+		// effected attrib location > 0
+		bool m_effectOnlyPoints;
+
+		bool m_effectAdd;
+		bool m_effectSubtract;
+		bool m_effectMultiply;
+		bool m_effectDivide;
+		bool m_effectPower;
+		bool m_effectLog;
+		bool m_effectSine;
+		bool m_effectClampLower;
+		bool m_effectClampUpper;
+
+		// stores m_automationModel->value(), used in updating
+		float m_bufferedAutomationValue;
+		// automation: connecting to floatmodels, -1 when it isn't conntected'
+		int m_automationModel;
+	};
+	// deletes the point's automation model
+	// if modelLocation == point location
+	void delAutomationModel(unsigned int modelLocation, bool callDataChanged);
+	// swapping values, "slide" moves the values (between) once left or right
+	// handle m_isFixedEndPoints when using this
+	void swap(unsigned int pointLocationA, unsigned int pointLocationB, bool slide);
```

Comment:
What does slide do exactly? (I haven't gotten to where it's defined yet, but this doesn't tell me what it does still)

Reply:
A better name would be shouldOffsetPoints, or something similar, this will be changed. It basically moves/shifts the points between the 2 location to keep the order (ordered by x coords).